tests/kernel/mellanox_config.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Mellanox Link protocol config
+# This configures the interfaces according to the
+# variable MLX_PROTOCOL. By default ETH if not set.
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use ipmi_backend_utils;
+
+our $device1 = '/dev/mst/mt4119_pciconf0';
+our $device2 = '/dev/mst/mt4119_pciconf0.1';
+
+sub show_links {
+    assert_script_run("mlxconfig -d $device1 q|grep LINK_TYPE");
+    assert_script_run("mlxconfig -d $device2 q|grep LINK_TYPE");
+}
+
+sub run {
+    use_ssh_serial_console;
+    my $mft_version = get_required_var('MFT_VERSION');
+    my $protocol = get_var('MLX_PROTOCOL') || 2;
+
+    zypper_call('ar -f -G ' . get_required_var('GA_REPO'));
```

Comment:
I wonder if we could add the GA_REPO in another way. While it is fine to install it here, I wonder if we could add it only depending on a variable being set outside any test?
I think it is better to have it here instead of the NFV and IB tests in the end, but maybe there is an even better place to add the REPO?

Reply:
Yes, I would prefer to have it outside. Maybe in the installation step, while configuring everything with yast or something similar. Not sure how, though.  Let's take this as a note for improvement later on.